dbt-athena/src/dbt/include/athena/macros/materializations/models/incremental/incremental.sql

```diff
@@ -62,6 +63,54 @@
     {%- endif -%}
     {% set build_sql = "select '" ~ query_result ~ "'" -%}
 
+  -- Running in full refresh, support High Availability for Iceberg table type --
+  -- Must use s3_data_naming schema_table_unique in order to support high availability --
+  -- on a full fresh for an incremental iceberg table --
+  {% elif should_full_refresh() and table_type == 'iceberg' and s3_data_naming == 'schema_table_unique' %}
```

Comment:
you might want to do a similar check to what was done [here](https://github.com/dbt-labs/dbt-adapters/blob/33b110c56fc401f0a476be4da51cbd8fad1e1e84/dbt-athena/src/dbt/include/athena/macros/materializations/models/table/table.sql#L38C9-L38C41)

`('unique' in s3_data_naming and external_location is none)`

pretty much we allow this feature by default for all unique location, and location that are not explicitly set via external_location.

Reply:
Makes sense to me.